### ISSUE TITLE
Re-sign third party frameworks

### ISSUE THREAD

**Issue by dvc94ch** [indygreg/PyOxidizer#538]
Currently it looks like rcodesign keeps the existing signature. I think the signature should be overwritten with the signing key?

**Comment by indygreg** [comment]
Framework signing has tons of bugs currently. I thought I tested this months ago but I apparently didn't.

I just pushed a pile of commits to make it behave a lot better. But there are still bugs, especially around nested bundle handling.

**Comment by indygreg** [comment]
I just pushed more commits that I think actually get framework and nested bundle signing into good shape. On `main`, I'm able to sign a macOS app bundle containing a nested framework. The framework signatures look nearly identical to what `codesign` is emitting. Non-main executables are getting registered in `CodeResources` correctly.

The only obvious difference in behavior I'm currently seeing is around the use of SHA-1 and SHA-256. `codesign` seems to be putting SHA-1 in the main code directory and putting SHA-256 in an alternative code directory.

I emulated this behavior for framework signing but I haven't yet done it for Mach-O binaries within bundles.

I'm honestly a bit confused by the behavior of `codesign`. I _thought_ Apple had done away with SHA-1 completely as SHA-256 has been supported for ages. But I could be wrong. It is possible they just always use SHA-1 + SHA-256 when signing bundles and I failed to realize this until recently.

We'll need to see what happens with notarization and possibly change defaults to always use SHA-1 + SHA-256.
